source/features/fork-source-link-same-view.tsx

```diff
@@ -8,36 +8,43 @@ import getDefaultBranch from '../github-helpers/get-default-branch';
 import {getRepo, getForkedRepo} from '../github-helpers';
 
 const isFilePath = (): boolean => pageDetect.isSingleFile()
-	|| (pageDetect.isRepoTree() && !pageDetect.isRepoRoot())
+	|| pageDetect.isRepoTree()
 	|| pageDetect.hasFileEditor();
 
-async function init(): Promise<void> {
+async function getEquivalentURL(): Promise<string> {
 	const forkedRepository = getRepo(getForkedRepo())!;
+	const defaultUrl = '/' + forkedRepository.nameWithOwner;
+
+	if (pageDetect.isConversation() || pageDetect.isRepoRoot()) {
+		// We must reset the link because the header is outside the ajaxed area
+		return defaultUrl;
+	}
+
 	const sameViewUrl = new GitHubURL(location.href).assign({
 		user: forkedRepository.owner,
 		repository: forkedRepository.name,
 	});
 
 	if (isFilePath()) {
 		sameViewUrl.branch = await getDefaultBranch(forkedRepository);
-	} else if (pageDetect.isIssue() || pageDetect.isPR()) {
-		sameViewUrl.assign({
-			route: '',
-			branch: '',
-			filePath: '',
-		});
+		if (!await doesFileExist(sameViewUrl)) {
+			return defaultUrl;
+		}
 	}
 
-	if (!isFilePath() || await doesFileExist(sameViewUrl)) {
-		select<HTMLAnchorElement>(`[data-hovercard-url="/${getForkedRepo()!}/hovercard"]`)!
-			.pathname = sameViewUrl.pathname;
-	}
+	return sameViewUrl.href;
+}
+
+async function init(): Promise<void> {
+	// The link must always be updated/reset. This pattern ensures that the link is always updated and never fails through some conditions.
+	select<HTMLAnchorElement>(`[data-hovercard-url="/${getForkedRepo()!}/hovercard"]`)!.href = await getEquivalentURL();
```

Comment:
I remember I updated this selector _somewhere_ since sometimes it doesn't match, I will try to find that